### PR TITLE
Fix failing k8s action in the workflow

### LINKED ISSUE #68: CI action error due to inaccessibility of k8s cluster
#### Description

Meshery SMP Action is failing on some CI runs due to inaccessibility of k8s cluster present in the workflow

#### Expected Behavior

The error must be rectified and CI run should be free of errors

### PR DESCRIPTION
**Pull request by alphaX86**
**Description**

This PR fixes #68 

**Notes for Reviewers**
WIP and test required

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by alphaX86** [comment]
Actions are being made to run with simultaneous changes. If worked, will ping for review

**Comment by hershd23** [comment]
Can you send the link to a local workflow run?

File: .github/workflows/configurable-benchmark-test-self-hosted.yaml [hunk 1]
```diff
@@ -83,14 +83,15 @@ jobs:
...
+          curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
+          sudo install minikube-linux-amd64 /usr/local/bin/minikube
 
-      - name: Setup Kubernetes
-        uses: manusa/actions-setup-minikube@v2.7.1
-        with:
-          minikube version: 'v1.23.2'
-          kubernetes version: 'v1.23.2'
-          driver: docker
-          start args: --disk-size 10000mb
```
Comment (hershd23) [inline .github/workflows/configurable-benchmark-test-self-hosted.yaml:93]: Any reason for replacing this action for a curl download then a manual install?

File: .github/workflows/configurable-benchmark-test.yaml [hunk 2]
```diff
@@ -36,12 +36,12 @@ jobs:
...
     steps:
       - name: Setup Kubernetes
-        uses: manusa/actions-setup-minikube@v2.7.1
+        uses: medyagh/setup-minikube@v0.0.8
         with:
           minikube version: v1.23.2 
           kubernetes version: v1.23.2
           driver: docker
-          start args: --disk-size 10000mb
+          memory: 5120m
```
Comment (hershd23) [inline .github/workflows/configurable-benchmark-test.yaml:44]: Why the change to a different workflow?

File: .github/workflows/scheduled-benchmarks-self-hosted.yaml [hunk 3]
```diff
@@ -90,12 +90,12 @@ jobs:
...
 
       - name: Setup Kubernetes
-        uses: manusa/actions-setup-minikube@v2.7.1
+        uses: manusa/actions-setup-minikube@v2.6.1
         with:
           minikube version: 'v1.23.2'
           kubernetes version: 'v1.23.2'
           driver: docker
-          start args: --disk-size 10000mb
+          start-args: --disk-size 10000mb
```
Comment (hershd23) [inline .github/workflows/scheduled-benchmarks-self-hosted.yaml:98]: Why the difference in the action each time while setting up kubernetes?

**Review by hershd23 — COMMENTED** [review]
There are 3 different ways we are setting up kubernetes in this PR

`manusa/actions-setup-minikube`
`medyagh/setup-minikube@v0.0.8`
And a manual curl get and install.

What is the problem with each of the methods exactly? Can we not use one method consistently throughout the workflows?

**Comment by alphaX86** [comment]
> There are 3 different ways we are setting up kubernetes in this PR
> 
> `manusa/actions-setup-minikube` `medyagh/setup-minikube@v0.0.8` And a manual curl get and install.
> 
> What is the problem with each of the methods exactly? Can we not use one method consistently throughout the workflows?

OK so the main issue is present in the self-hosted runner where the k8s Workflow didn't work during setup and the action throws an error saying there's no permission for the action to unzip files during setup. This is why for self-hosted runners alone, I used `curl` based method and eventually it worked rather than using other 2 k8s action which throws several access errors.

For normal ones, I just changed them for testing which is reliable. Don't get alarmed 😅

File: .github/workflows/configurable-benchmark-test-self-hosted.yaml (hunk 1 above)
Comment (alphaX86) [reply .github/workflows/configurable-benchmark-test-self-hosted.yaml:93]: Self-hosted runners didn't allow the action to setup k8s, so curl method works for now

File: .github/workflows/configurable-benchmark-test.yaml (hunk 2 above)
Comment (alphaX86) [reply .github/workflows/configurable-benchmark-test.yaml:44]: I'll revert it, no issues spotted here actually

File: .github/workflows/scheduled-benchmarks-self-hosted.yaml (hunk 3 above)
Comment (alphaX86) [reply .github/workflows/scheduled-benchmarks-self-hosted.yaml:98]: This one didn't work too, I'll change it to curl based method

**Comment by alphaX86** [comment]
> Can you send the link to a local workflow run?

All workflows are run here itself in this repo, so the logs are present in the Actions tab

**Comment by gyohuangxin** [comment]
Excellent! It seems your changes to kubernetes worked: https://github.com/layer5io/meshery-smp-action/actions/runs/3472699467/jobs/5803912095, so please revise this PR based on @hershd23 's comments.

**Comment by alphaX86** [comment]
Logs:
https://github.com/layer5io/meshery-smp-action/actions/runs/3573114008/jobs/6006793795

https://github.com/layer5io/meshery-smp-action/actions/runs/3573116243

**Comment by leecalcote** [comment]
/home/runner/work/_actions/layer5io/meshery-smp-action/master/mesheryctl.sh: line 33: adapters: bad array subscript
[1206](https://github.com/layer5io/meshery-smp-action/actions/runs/3573114008/jobs/6006793795#step:5:1207)

**Comment by gyohuangxin** [comment]
> `/home/runner/work/_actions/layer5io/meshery-smp-action/master/mesheryctl.sh: line 33: adapters: bad array subscript` [1206](https://github.com/layer5io/meshery-smp-action/actions/runs/3573114008/jobs/6006793795#step:5:1207)

@alphaX86 Are you looking at this error or do you need some help?

**Comment by alphaX86** [comment]
> > `/home/runner/work/_actions/layer5io/meshery-smp-action/master/mesheryctl.sh: line 33: adapters: bad array subscript` [1206](https://github.com/layer5io/meshery-smp-action/actions/runs/3573114008/jobs/6006793795#step:5:1207)
> 
> @alphaX86 Are you looking at this error or do you need some help?

Looking for now, got my laptop recently fixed after a week :sweat_smile:

**Comment by gyohuangxin** [comment]
@alphaX86 Hi, do you have any update?

**Comment by alphaX86** [comment]
Uh oh! 🙆‍♂️ totally forgot this... I'll resume this work on Next week Mon and update you back... Until then, this PR is now a draft

**Comment by gyohuangxin** [comment]
@alphaX86 Good, it's ok. Thank you!

**Review by gyohuangxin — APPROVED** [review]
Thanks for continuing to fix this issue, lgtm.

**Review by hershd23 — APPROVED** [review]
LGTM

**Comment by gyohuangxin** [comment]
@hershd23 Thanks.
